.ci/docker/manywheel/build.sh

```diff
@@ -140,7 +140,7 @@ fi
         "${TOPDIR}/.ci/docker/"
 )
 
-GITHUB_REF=${GITHUB_REF:-$(git symbolic-ref -q HEAD || git describe --tags --exact-match)}
```

Comment:
It would be good to have a bit more history in PR description why this was added and why this is no longer relevant

IMO this logic is prone to error and if we expect this script to always be invoked with valid GITHUB_REF environment variable defined, than imo it's better to error out that keep `-dev` suffix